nestkernel/event_delivery_manager.cpp

```diff
@@ -356,7 +383,21 @@ EventDeliveryManager::gather_spike_data_( const thread tid,
       // Needs to be called /after/ set_end_and_invalid_markers_.
       set_complete_marker_spike_data_( assigned_ranks, send_buffer_position, send_buffer );
 #pragma omp barrier
+      // TODO: Discuss, if the barrier above should be better placed after if-clause
+    }
+
+#ifdef TIMER_DETAILED
+#pragma omp barrier
+    if ( tid == 0 )
+    {
+      sw_collocate_spike_data_.stop();
+
+      // TODO_SDR: Following function call commented out because it causes real runs to hang!
+      // kernel().mpi_manager.synchronize(); // to get an accurate time measurement
+      // across ranks
```

Comment:
Can this be removed?

Reply:
I agree, we shouldn't introduce a MPI synchronization point. This would yield significant overhead, which is also not accounted for in any timer at the moment.

Reply:
Agreed. Now removed synch point and comment.